blocks/text.ts

```diff
@@ -828,15 +838,15 @@ const TEXT_JOIN_MUTATOR_MIXIN = {
     this.updateShape_();
     // Reconnect any child blocks.
     for (let i = 0; i < this.itemCount_; i++) {
-      Mutator.reconnect(connections[i], this, 'ADD' + i);
+      Mutator.reconnect(connections[i]!, this, 'ADD' + i);
     }
   },
   /**
    * Store pointers to any connected child blocks.
-   * @param {!Block} containerBlock Root block in mutator.
-   * @this {Block}
+   *
+   * @param containerBlock Root block in mutator.
    */
-  saveConnections: function(containerBlock) {
+  saveConnections: function(this: JoinMutatorBlock, containerBlock: Block) {
```

Comment:
Again: a more specific type for `containerBlock` should be possible and might be useful.

Reply:
I'm neutral on the specific type for the container block. Do you want me to add it? My default is to leave it alone.